webapp/channels/src/components/gif_picker/components/gif_picker_items.tsx

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useMemo} from 'react';
+import {GiphyFetch} from '@giphy/js-fetch-api';
+import {EmojiVariationsListProps, Grid} from '@giphy/react-components';
+
+const giphyFetch = new GiphyFetch('DxfEk2t4CR8bv4A1kviDBRMP9i1og3Da');
+
+const GUTTER_BETWEEN_GIFS = 8;
+const NUM_OF_GIFS_COLUMNS = 2;
+const GIF_DEFAULT_WIDTH = 350;
+const GIF_MARGIN_ENDS = 12;
+
+interface Props {
+    filter: string;
+    onClick: EmojiVariationsListProps['onGifClick'];
+}
+
+function GifPickerItems(props: Props) {
+    function fetch(offset: number) {
+        // We dont have to throttled the fetching as the library does it for us
+        if (props.filter.length > 0) {
+            return giphyFetch.search(props.filter, {offset, limit: 10});
+        }
+
+        return giphyFetch.trending({offset, limit: 10});
+    }
+
+    const width = useMemo(() => {
```

Comment:
I think `useMemo` is called while React is making an update which seems like it could cause problems. Are you sure that this is always called when the `emoji-picker-tabs` exists on the DOM?

Between this and my comment below, maybe we should just hardcode this value. 0/5 on that though

Reply:
This use memo, specially looking at the dependencies, will run only during the first render, before rendering anything. So... if `emoji-picker-tabs` does not exist during the first render, it will use the `GIF_DEFAULT_WIDTH`. So... where exactly does this `emoji-picker-tabs` exist in time an space? Below this component (and therefore will never exist at this point) or above? Rendering at the same time or after (and therefore will not exist at this point) or always before this component?

Reply:
i read about this for a bit, it seems that `getboundingclientrect` when called in a loop forces a browser reflow. Looping could occur if width is not memorized. WDUT?

Reply:
Oh yeah, calling it only once in a `useMemo` is definitely preferable to that, but I meant that this is being called during the functional component version of `render` which seems minorly incorrect. I can't think of a better time to call it in this case though since you need the width synchronously, so this is probably good as-is